samza-core/src/main/java/org/apache/samza/execution/JobGraphJsonGenerator.java

```diff
@@ -39,69 +40,85 @@
 /**
  * This class generates the JSON representation of the {@link JobGraph}.
  */
-public class JobGraphJsonGenerator {
+/* package private */ class JobGraphJsonGenerator {
 
   /**
    * This class provides the necessary connection of operators for traversal.
    */
   static abstract class Traversable {
-    @JsonProperty("NextOperatorIds")
+    @JsonProperty("nextOperatorIds")
     Set<Integer>  nextOperatorIds = new HashSet<>();
   }
 
   static final class OperatorJson extends Traversable {
-    @JsonProperty("OpCode")
+    @JsonProperty("opCode")
     String opCode;
-    @JsonProperty("OpId")
+    @JsonProperty("opId")
     int opId;
-    @JsonProperty("OutputStreamId")
+    @JsonProperty("outputStreamId")
     String outputStreamId;
-    @JsonProperty("PairedOpId")
+    @JsonProperty("pairedOpId")
     int pairedOpId = -1;  //for join operator, we will have a pair nodes for two partial joins
+    @JsonProperty("caller")
+    String caller;
   }
 
   static final class StreamSpecJson {
-    @JsonProperty("Id")
+    @JsonProperty("id")
     String id;
-    @JsonProperty("SystemName")
+    @JsonProperty("systemName")
     String systemName;
-    @JsonProperty("PhysicalName")
+    @JsonProperty("physicalName")
     String physicalName;
-    @JsonProperty("PartitionCount")
+    @JsonProperty("partitionCount")
     int partitionCount;
   }
 
   static final class StreamEdgeJson {
-    @JsonProperty("StreamSpec")
+    @JsonProperty("streamSpec")
     StreamSpecJson streamSpec;
+    @JsonProperty("sourceJobs")
+    List<String> sourceJobs;
+    @JsonProperty("targetJobs")
+    List<String> targetJobs;
   }
 
   static final class OperatorGraphJson {
-    @JsonProperty("InputStreams")
+    @JsonProperty("inputStreams")
     List<InputStreamJson> inputStreams;
-    @JsonProperty("Operators")
+    @JsonProperty("operators")
     Map<Integer, OperatorJson> operators = new HashMap<>();
+    @JsonProperty("outputStreams")
+    List<String> outputStreams;
   }
 
   static final class InputStreamJson extends Traversable {
-    @JsonProperty("StreamId")
+    @JsonProperty("streamId")
```

Comment:
Same comment as previous PR: Input and output streams should be symmetric. Currently InputStreamJson has nextOperatorIds but outputStreamIds are part of the OperatorJson.